MapzenSDK/Themes.swift

```diff
@@ -10,149 +10,224 @@ import Foundation
 
 /**
  This is the foundational code for custom stylesheet and theme support. This area will likely be the focus of updates & changes over the next several releases, so we would recommend avoiding implementing your own custom stylesheet classes until we have further vetted this implementation. Documentation will be written once we've solidified on the protocol requirements and implementation details.
- 
+
  We do however welcome suggestions / improvements to this API on our github at https://github.com/mapzen/ios
  */
 
 
-public protocol StyleSheet : class {
-
-  var fileLocation: URL? { get }
-  var remoteFileLocation: URL? { get }
-  static var stylesheetRoot: String { get }
-  var appliedTheme: Theme { get }
-  var houseStylesRoot: String { get }
-  static var styleSheetFileName: String { get }
-  var importString: String { get }
-  var relativePath: String { get }
-  var mapStyle: MapStyle? { get set }
+@objc public protocol StyleSheet : class {
+
+  @objc var fileLocation: URL? { get }
+  @objc var remoteFileLocation: URL? { get }
+  @objc var styleSheetRoot: String { get }
+  @objc var houseStylesRoot: String { get }
+  @objc var styleSheetFileName: String { get }
+  @objc var importString: String { get }
+  @objc var relativePath: String { get }
+  @objc var mapStyle: MapStyle { get set }
```

Comment:
Do we need this enum still? Or do you want to deprecate it and remove it in a later release?

If we keep it, should it be a get-only property so that something like:
```swift
let bubblewrap = BubbleWrapStyle()
bubblewrap.mapStyle = .cinnabar
```
isn't possible

Reply:
Duh that makes sense.